sdk/attestation/azure-security-attestation/azure/security/attestation/_administration_client.py

```diff
@@ -4,316 +4,680 @@
+            raise TypeError('add_policy_management_certificate takes a single positional parameter. found {}'.format(len(args)))
+        certificate_to_add = args[0]
+
+        signing_key = kwargs.pop("signing_key", self._signing_key)
+        signing_certificate = kwargs.pop(
+            "signing_certificate", self._signing_certificate
+        )
+
+        if not signing_key or not signing_certificate:
+            raise ValueError(
+                "A signing certificate and key must be provided to add_policy_management_certificate."
+            )
+
+        # Verify that the provided certificate is a valid PEM encoded X.509 certificate
+        certificate_to_add = load_pem_x509_certificate(certificate_to_add.encode("ascii"))
+
+        jwk = JSONWebKey(
+            kty="RSA",
+            x5_c=[
+                base64.b64encode(
+                    certificate_to_add.public_bytes(serialization.Encoding.DER)
+                ).decode("ascii")
+            ],
+        )
+        add_body = AttestationCertificateManagementBody(policy_certificate=jwk)
+        cert_add_token = AttestationToken(
             body=add_body,
-            signer=signing_key,
-            body_type=AttestationCertificateManagementBody)
+            signing_key=signing_key,
+            signing_certificate=signing_certificate,
+            body_type=AttestationCertificateManagementBody,
+        )
+
+        # Merge our existing config options with the options for this API call.
+        # Note that this must be done before calling into the implementation
+        # layer because the implementation layer doesn't like keyword args that
+        # it doesn't expect :(.
+        options = merge_validation_args(self._config._kwargs, kwargs)
+
+        cert_response = self._client.policy_certificates.add(
+            cert_add_token.to_jwt_string(), **kwargs
+        )
+        token = AttestationToken(
+            token=cert_response.token,
+            body_type=GeneratedPolicyCertificatesModificationResult,
+        )
 
-        cert_response = self._client.policy_certificates.add(cert_add_token.serialize(), **kwargs)
-        token = AttestationToken[GeneratedPolicyCertificatesModificationResult](token=cert_response.token,
-            body_type=GeneratedPolicyCertificatesModificationResult)
-        if self._config.token_validation_options.validate_token:
-            if not token.validate_token(self._config.token_validation_options, self._get_signers(**kwargs)):
-                raise Exception("Token Validation of PolicyCertificate Add API failed.")
-        return AttestationResponse[PolicyCertificatesModificationResult](token, PolicyCertificatesModificationResult._from_generated(token.get_body()))
+        if options.get("validate_token", True):
+            token._validate_token(self._get_signers(**kwargs), **options)
+        return (
+            AttestationPolicyCertificateResult._from_generated(token._get_body()),
```

Comment:
I would imagine you'll need a `pylint: disable = protected-access` at the end of this line.

Reply:
pylint seems ok with it as does tox lint. So does the CI pipeline.

What am I doing wrong here?

Reply:
I think pylint's not working as expected in CI, but you can run the correct checks using the instructions in [this comment](https://github.com/Azure/azure-sdk-for-python/pull/19164/files#r661678448)

Reply:
tox lint was ok, pylint not so much.

Fixing the hundreds of errors...